### DISCUSSION TITLE
Feature: Allow comments for routes in CP

### DISCUSSION THREAD

**Discussion by jkorff** [craftcms/cms#8583] [Ideas]
### Description
When setting up routes in the CP (and not the config file) I am unable to annotate the route to explain its existence.
This is especially helpful if there is more than one user / editor, but also to jog my own memory after time.

### Steps to reproduce

1. Craft > Settings > Routes
2. Click on any route: A plain text field could show below the "Load this template" box.

### Additional info

- Craft version: 3.0.21
- PHP version:
- Database driver & version:
- Plugins & versions:

**Comment by brandonkelly** [comment]
For now, this could be done by adding a comment in the route’s template file.

**Comment by jkorff** [comment]
You mean the "Load this template" target?

**Comment by brandonkelly** [comment]
Well, in the actual template file that you reference in that setting.

For example if you have a route that loads the `blog/_archive` template, then you could put at the top of your `blog/_archive.twig` file something like

```twig
{# This route will get loaded for `blog/archive/<Year>` requests.
   ... whatever other info is relevant here ...
   #}
```

**Comment by jkorff** [comment]
Yes, that's what I thought, thank you for clarifying.

It might be a better idea to put most routing into the route.php config file? That way documentation and routes stay together. Is there a preferred way for when the config file should be used? (Other than hide these from users.)

**Comment by brandonkelly** [comment]
If you’re comfortable with PHP and regular expressions, routes.php is probably better, since you have more control there.

**Comment by CreateSean** [comment]
I'd also like to see the option to add comments in the routes section of the CP.

**Comment by brandonkelly** [comment]
@CreateSean Best way to cast your vote is to add a 👍 reaction to the initial post.
